mmcls/models/backbones/mlp_mixer.py

```diff
@@ -0,0 +1,198 @@
+            dim,
+            channels_dim,
+            act_layer_cfg=act_layer_cfg,
+            drop_rate=drop_rate)
+
+    def __build_mlp(self,
+                    in_features,
+                    hidden_features=None,
+                    out_features=None,
+                    act_layer_cfg=None,
+                    drop_rate=0.):
+        out_features = out_features or in_features
+        hidden_features = hidden_features or in_features
+        mlp = Sequential(*[
+            nn.Linear(in_features, hidden_features),
+            build_activation_layer(act_layer_cfg),
+            nn.Dropout(drop_rate),
+            nn.Linear(hidden_features, out_features),
+            nn.Dropout(drop_rate)
+        ])
+
+        return mlp
+
+    def forward(self, x):
+        x = x + self.drop_path(
+            self.mlp_tokens(self.norm1(x).transpose(1, 2)).transpose(1, 2))
+        x = x + self.drop_path(self.mlp_channels(self.norm2(x)))
+        return x
+
+
+@BACKBONES.register_module()
+class MlpMixer(BaseModule):
+    """MlpMixer backbone.
+
+    Pytorch implementation of `MLP-Mixer: An all-MLP Architecture for Vision
+    <https://arxiv.org/pdf/2105.01601.pdf>`_
+
+    Some code is borrowed and modified from timm.
+
+    Args:
+        img_size (int): Input image size. Default to 224.
+        in_channels (int): Number of input channels. Default to 3.
+        patch_size (int): Size of patch. Default to 16.
+        num_blocks (int): Num of mixer block. Default to 8.
+        embed_dims (int): Embedding dimension.
+        mlp_ratio (tuple | int): Multiplier for the D_s and D_c. Default
+            to (0.5, 4.0)
+        norm_cfg (dict): Configuration for the normalization layer. Default to
+            dict(type='LN', eps=1e-6).
+        act_layer_cfg (dict): Configuration for the activation layer. Default
+            to dict(type='GELU').
+        drop_rate (float): Dropout rate for the dropout layer in MixerBlock.
+            Default to 0.
+        drop_path_rate (float | list): Drop path rate for the DropPath (
+            Stochastic Depth) in Mixer. Default to 0.
+        stem_norm (bool): If use normalization in the stem part. Default to
+            False.
+    """
+
+    def __init__(
+        self,
+        img_size=224,
+        in_channels=3,
+        patch_size=16,
+        num_blocks=8,
+        embed_dims=512,
+        mlp_ratio=(0.5, 4.0),
+        norm_cfg=dict(type='LN', eps=1e-6),
+        act_layer_cfg=dict(type='GELU'),
+        drop_rate=0.,
+        drop_path_rate=0.,
+        stem_norm=False,
+    ):
+        super().__init__()
+
+        self.num_features = self.embed_dims = embed_dims
+
+        self.stem = PatchEmbed(
+            img_size=img_size,
+            conv_cfg=dict(kernel_size=patch_size, stride=patch_size),
+            in_channels=in_channels,
+            embed_dims=embed_dims,
+            norm_cfg=norm_cfg if stem_norm else None)
+
+        if isinstance(drop_path_rate, (int, float)):
```

Comment:
`drop_path_rate` only accepts `float` or `list`

Reply:
Should we support `int` since users may accidentally pass `0`.